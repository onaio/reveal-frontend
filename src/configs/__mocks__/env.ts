/** You can hard code variables that would have been
 * extracted from the environment here, and then mock
 * this file from your test file.
 * The goal is to have consistent test results
 */

export const SUPERSET_PLANS_SLICE = 0;
export type SUPERSET_PLANS_SLICE = typeof SUPERSET_PLANS_SLICE;

export const SUPERSET_MAX_RECORDS = 3000;
export type SUPERSET_MAX_RECORDS = typeof SUPERSET_MAX_RECORDS;

export const GISIDA_TIMEOUT = 3000;
export type GISIDA_TIMEOUT = typeof GISIDA_TIMEOUT;

export const GISIDA_MAPBOX_TOKEN = 'hunter2';
export type GISIDA_MAPBOX_TOKEN = typeof GISIDA_MAPBOX_TOKEN;

export const GISIDA_ONADATA_API_TOKEN = 'hunter2';
export type GISIDA_ONADATA_API_TOKEN = typeof GISIDA_ONADATA_API_TOKEN;

export const DIGITAL_GLOBE_CONNECT_ID = 'hunter2';
export type DIGITAL_GLOBE_CONNECT_ID = typeof DIGITAL_GLOBE_CONNECT_ID;

export const OPENSRP_API_BASE_URL = 'https://test.smartregister.org/opensrp/rest/';
export type OPENSRP_API_BASE_URL = typeof OPENSRP_API_BASE_URL;

export const PLAN_UUID_NAMESPACE = '1bb3c217-af1c-4749-b3df-4f6108582e3c';
export type PLAN_UUID_NAMESPACE = typeof PLAN_UUID_NAMESPACE;

export const ACTION_UUID_NAMESPACE = '1bb3c217-af1c-4749-b3df-4f6108582e3c';
export type ACTION_UUID_NAMESPACE = typeof ACTION_UUID_NAMESPACE;

export const GA_CODE = 'UA-154169046-1';
export type GA_CODE = typeof GA_CODE;

export const GA_ENV = 'test';
export type GA_ENV = typeof GA_ENV;

export const DEFAULT_PLAN_VERSION = '2';
export type DEFAULT_PLAN_VERSION = typeof DEFAULT_PLAN_VERSION;

export const SUPERSET_JURISDICTIONS_SLICE = 1;
export type SUPERSET_JURISDICTIONS_SLICE = typeof SUPERSET_JURISDICTIONS_SLICE;

export const SUPERSET_STRUCTURES_SLICE = 2;
export type SUPERSET_STRUCTURES_SLICE = typeof SUPERSET_STRUCTURES_SLICE;

export const SUPERSET_GOALS_SLICE = 3;
export type SUPERSET_GOALS_SLICE = typeof SUPERSET_GOALS_SLICE;

export const SUPERSET_TASKS_SLICE = 4;
export type SUPERSET_TASKS_SLICE = typeof SUPERSET_TASKS_SLICE;

/** Do you want to enable team assignment? */
export const ENABLE_ASSIGN = true;

/** Do you want to enable the IRS features? */
export const ENABLE_IRS = true;

/** Do you want to enable the Focus Investigation features? */
export const ENABLE_FI = true;

export const ENABLE_PRACTITIONERS = true;

/** Do you want to enable the users page? */
export const ENABLE_USERS = true;

/** Do you want to enable the about page? */
export const ENABLE_ABOUT = true;

/** Do you want to enable views dealing with teams */
export const ENABLE_TEAMS = true;

export const SUPERSET_IRS_REPORTING_JURISDICTIONS_DATA_SLICES = '11,12';
export type SUPERSET_IRS_REPORTING_JURISDICTIONS_DATA_SLICES = typeof SUPERSET_IRS_REPORTING_JURISDICTIONS_DATA_SLICES;
export const SUPERSET_IRS_REPORTING_PLANS_SLICE = '13';
export type SUPERSET_IRS_REPORTING_PLANS_SLICE = typeof SUPERSET_IRS_REPORTING_PLANS_SLICE;
export const SUPERSET_IRS_REPORTING_STRUCTURES_DATA_SLICE = '14';
export type SUPERSET_IRS_REPORTING_STRUCTURES_DATA_SLICE = typeof SUPERSET_IRS_REPORTING_STRUCTURES_DATA_SLICE;
export const SUPERSET_IRS_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL = '4';
export type SUPERSET_IRS_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL = typeof SUPERSET_IRS_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL;
export const SUPERSET_IRS_REPORTING_JURISDICTIONS_COLUMNS = 'zambiaJurisdictions2019';
export type SUPERSET_IRS_REPORTING_JURISDICTIONS_COLUMNS = typeof SUPERSET_IRS_REPORTING_JURISDICTIONS_COLUMNS;
export const SUPERSET_IRS_REPORTING_FOCUS_AREAS_COLUMNS = 'zambiaFocusArea2019';
export type SUPERSET_IRS_REPORTING_FOCUS_AREAS_COLUMNS = typeof SUPERSET_IRS_REPORTING_FOCUS_AREAS_COLUMNS;
export const SUPERSET_IRS_REPORTING_INDICATOR_STOPS = 'zambia2019';
export type SUPERSET_IRS_REPORTING_INDICATOR_STOPS = typeof SUPERSET_IRS_REPORTING_INDICATOR_STOPS;
export const SUPERSET_IRS_REPORTING_INDICATOR_ROWS = 'zambia2019';
export type SUPERSET_IRS_REPORTING_INDICATOR_ROWS = typeof SUPERSET_IRS_REPORTING_INDICATOR_ROWS;

export const DATE_FORMAT = 'yyyy-MM-dd';
export type DATE_FORMAT = typeof DATE_FORMAT;

export const DEFAULT_TIME = 'T00:00:00+00:00';
export type DEFAULT_TIME = typeof DEFAULT_TIME;

export const DEFAULT_PLAN_DURATION_DAYS = 20;
export type DEFAULT_PLAN_DURATION_DAYS = typeof DEFAULT_PLAN_DURATION_DAYS;

export const DEFAULT_ACTIVITY_DURATION_DAYS = 7;
export type DEFAULT_ACTIVITY_DURATION_DAYS = typeof DEFAULT_ACTIVITY_DURATION_DAYS;

export const USERS_REQUEST_PAGE_SIZE = 51;
export type USERS_REQUEST_PAGE_SIZE = typeof USERS_REQUEST_PAGE_SIZE;

export const LANGUAGE = 'en';
export type LANGUAGE = typeof LANGUAGE;

export const OAUTH_GET_STATE_URL = 'http://localhost:3000/oauth/state';
export type OAUTH_GET_STATE_URL = typeof OAUTH_GET_STATE_URL;

export const DISABLE_LOGIN_PROTECTION = false;

export const EXPRESS_OAUTH_GET_STATE_URL = 'http://localhost:3000/oauth/state';

export const EXPRESS_OAUTH_LOGOUT_URL = 'http://localhost:3000/logout';

export const BACKEND_ACTIVE = true;

export const SUPERSET_JURISDICTIONS_DATA_SLICE = 471;
export type SUPERSET_JURISDICTIONS_DATA_SLICE = typeof SUPERSET_JURISDICTIONS_DATA_SLICE;

export const SUPERSET_MDA_POINT_REPORTING_JURISDICTIONS_DATA_SLICES = '12,esw-jurisdictions';
export type SUPERSET_MDA_POINT_REPORTING_JURISDICTIONS_DATA_SLICES = typeof SUPERSET_MDA_POINT_REPORTING_JURISDICTIONS_DATA_SLICES;

export const SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE = '01';
export type SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE = typeof SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE;

export const SUPERSET_MDA_POINT_REPORTING_JURISDICTIONS_COLUMNS = '0';
export type SUPERSET_MDA_POINT_REPORTING_JURISDICTIONS_COLUMNS = typeof SUPERSET_MDA_POINT_REPORTING_JURISDICTIONS_COLUMNS;

export const SUPERSET_MDA_POINT_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL = '2';
export type SUPERSET_MDA_POINT_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL = typeof SUPERSET_MDA_POINT_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL;

export const SUPERSET_MDA_POINT_REPORTING_FOCUS_AREAS_COLUMNS = '0';
export type SUPERSET_MDA_POINT_REPORTING_FOCUS_AREAS_COLUMNS = typeof SUPERSET_MDA_POINT_REPORTING_FOCUS_AREAS_COLUMNS;
export const REACT_APP_VERSION = 'v0.4.2';

export const ENABLED_PLAN_TYPES = ['FI', 'IRS', 'MDA', 'MDA-Point'];
export type ENABLED_PLAN_TYPES = typeof ENABLED_PLAN_TYPES;

export const ENABLED_FI_REASONS = ['Case Triggered', 'Routine'];
export type ENABLED_FI_REASONS = typeof ENABLED_FI_REASONS;

export const WEBSITE_NAME = 'Reveal';

export const NAVBAR_BRAND_IMG_SRC =
  'https://github.com/onaio/reveal-frontend/raw/b0756b0c1ed95dc6db23dc4fd1375c81bdf7fb1b/src/assets/images/thai-logo.png';

export const REVEAL_BRAND_IMG_SRC =
  'https://github.com/onaio/reveal-frontend/raw/b0756b0c1ed95dc6db23dc4fd1375c81bdf7fb1b/src/assets/images/logo.png';

export const ENABLE_MDA_POINT = true;

export const SUPERSET_MDA_POINT_LOCATION_REPORT_DATA_SLICE = '01';

export const SHOW_MDA_SCHOOL_REPORT_LABEL = true;

export const HIDDEN_MAP_LEGEND_ITEMS = [];

export const HIDDEN_PLAN_STATUSES = ['retired'];

export const ENABLE_JURISDICTIONS_AUTO_SELECTION = true;

export const ENABLE_JURISDICTION_AUTO_SELECTION_FOR_PLAN_TYPES = 'IRS,Dynamic-IRS'.split(',');

export const SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE = '1337';
export const SUPERSET_DYNAMIC_MDA_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL = '2';
export const SUPERSET_DYNAMIC_MDA_REPORTING_FOCUS_AREAS_COLUMNS = 'zambiaLowerUpper2020';
export const SUPERSET_DYNAMIC_MDA_REPORTING_JURISDICTIONS_COLUMNS = 'zambiaMDAUpper2020';
export const SUPERSET_DYNAMIC_MDA_REPORTING_JURISDICTIONS_DATA_SLICES = '1338';
export const ENABLE_DYNAMIC_MDA = true;

export const ENABLED_JURISDICTION_METADATA_IDENTIFIER_OPTIONS = 'COVERAGE,POPULATION,RISK,TARGET'.split(
  ','
);

export const ASSIGNMENT_PAGE_SHOW_MAP = false;
