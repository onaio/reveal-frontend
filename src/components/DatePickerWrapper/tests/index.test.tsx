import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Field, FieldProps, FormikHandlers } from 'formik';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import DatePickerWrapper from '..';
import { DATE_FORMAT } from '../../../configs/env';
import props from './fixtures';

const history = createBrowserHistory();

describe('components/DatePickerWrapper', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const handleChange = () => jest.fn();
    shallow(<DatePickerWrapper selected={new Date()} onChange={handleChange} {...props} />);
  });
});
