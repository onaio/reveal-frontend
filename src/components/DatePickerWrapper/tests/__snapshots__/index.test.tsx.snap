// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/DatePickerWrapper onHandle change is called when date is chosen 1`] = `
<DatePickerWrapper
  className="form-control"
  dateFormat="yyyy-MM-dd"
  disabled={false}
  field={
    Object {
      "name": "start",
      "onChange": [MockFunction],
      "value": "2019-08-05T10:17:17.765Z",
    }
  }
  form={
    Object {
      "dirty": false,
      "errors": Object {},
      "initialValues": Object {
        "activities": Array [
          Object {
            "actionCode": "Case Confirmation",
            "actionDescription": "Confirm the index case",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Case Confirmation",
            "goalDescription": "Confirm the index case",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 1,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "RACD Register Family",
            "actionDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Family Registration",
            "goalDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 100,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "Blood Screening",
            "actionDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Blood screening",
            "goalDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 100,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "Bednet Distribution",
            "actionDescription": "Visit 100% of residential structures in the operational area and provide nets",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Bednet Distribution",
            "goalDescription": "Visit 100% of residential structures in the operational area and provide nets",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 100,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "Larval Dipping",
            "actionDescription": "Perform a minimum of three larval dipping activities in the operational area",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Larval Dipping",
            "goalDescription": "Perform a minimum of three larval dipping activities in the operational area",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 3,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "Mosquito Collection",
            "actionDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Mosquito Collection",
            "goalDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 3,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "BCC",
            "actionDescription": "Conduct BCC activity",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Behaviour Change Communication",
            "goalDescription": "Complete at least 1 BCC activity for the operational area",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 1,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
        ],
        "caseNum": "",
        "date": "2019-08-05T10:17:17.765Z",
        "end": "2019-08-25T10:17:17.765Z",
        "identifier": "",
        "interventionType": "FI",
        "jurisdictions": Array [
          Object {
            "id": "",
            "name": "",
          },
        ],
        "name": "",
        "start": "2019-08-05T10:17:17.765Z",
        "status": "draft",
        "title": "",
        "version": "1",
      },
      "isSubmitting": false,
      "isValid": false,
      "isValidating": false,
      "submitCount": 0,
      "touched": Object {},
      "validateOnBlur": true,
      "validateOnChange": true,
      "values": Object {
        "activities": Array [
          Object {
            "actionCode": "Case Confirmation",
            "actionDescription": "Confirm the index case",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Case Confirmation",
            "goalDescription": "Confirm the index case",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 1,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "RACD Register Family",
            "actionDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Family Registration",
            "goalDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 100,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "Blood Screening",
            "actionDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Blood screening",
            "goalDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 100,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "Bednet Distribution",
            "actionDescription": "Visit 100% of residential structures in the operational area and provide nets",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Bednet Distribution",
            "goalDescription": "Visit 100% of residential structures in the operational area and provide nets",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 100,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "Larval Dipping",
            "actionDescription": "Perform a minimum of three larval dipping activities in the operational area",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Larval Dipping",
            "goalDescription": "Perform a minimum of three larval dipping activities in the operational area",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 3,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "Mosquito Collection",
            "actionDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Mosquito Collection",
            "goalDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 3,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
          Object {
            "actionCode": "BCC",
            "actionDescription": "Conduct BCC activity",
            "actionIdentifier": "",
            "actionReason": "Investigation",
            "actionTitle": "Behaviour Change Communication",
            "goalDescription": "Complete at least 1 BCC activity for the operational area",
            "goalDue": "2019-08-12T10:17:17.765Z",
            "goalPriority": "medium-priority",
            "goalValue": 1,
            "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
            "timingPeriodStart": "2019-08-05T10:17:17.765Z",
          },
        ],
        "caseNum": "",
        "date": "2019-08-05T10:17:17.765Z",
        "end": "2019-08-25T10:17:17.765Z",
        "identifier": "",
        "interventionType": "FI",
        "jurisdictions": Array [
          Object {
            "id": "",
            "name": "",
          },
        ],
        "name": "",
        "start": "2019-08-05T10:17:17.765Z",
        "status": "draft",
        "title": "",
        "version": "1",
      },
    }
  }
  id="start"
  onChange={[MockFunction]}
  required={true}
  selected={2019-08-05T12:12:25.670Z}
  type="date"
>
  <DatePicker
    allowSameDay={false}
    className="form-control"
    dateFormat="yyyy-MM-dd"
    dateFormatCalendar="LLLL yyyy"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    field={
      Object {
        "name": "start",
        "onChange": [MockFunction],
        "value": "2019-08-05T10:17:17.765Z",
      }
    }
    form={
      Object {
        "dirty": false,
        "errors": Object {},
        "initialValues": Object {
          "activities": Array [
            Object {
              "actionCode": "Case Confirmation",
              "actionDescription": "Confirm the index case",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Case Confirmation",
              "goalDescription": "Confirm the index case",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 1,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "RACD Register Family",
              "actionDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Family Registration",
              "goalDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 100,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "Blood Screening",
              "actionDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Blood screening",
              "goalDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 100,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "Bednet Distribution",
              "actionDescription": "Visit 100% of residential structures in the operational area and provide nets",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Bednet Distribution",
              "goalDescription": "Visit 100% of residential structures in the operational area and provide nets",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 100,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "Larval Dipping",
              "actionDescription": "Perform a minimum of three larval dipping activities in the operational area",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Larval Dipping",
              "goalDescription": "Perform a minimum of three larval dipping activities in the operational area",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 3,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "Mosquito Collection",
              "actionDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Mosquito Collection",
              "goalDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 3,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "BCC",
              "actionDescription": "Conduct BCC activity",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Behaviour Change Communication",
              "goalDescription": "Complete at least 1 BCC activity for the operational area",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 1,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
          ],
          "caseNum": "",
          "date": "2019-08-05T10:17:17.765Z",
          "end": "2019-08-25T10:17:17.765Z",
          "identifier": "",
          "interventionType": "FI",
          "jurisdictions": Array [
            Object {
              "id": "",
              "name": "",
            },
          ],
          "name": "",
          "start": "2019-08-05T10:17:17.765Z",
          "status": "draft",
          "title": "",
          "version": "1",
        },
        "isSubmitting": false,
        "isValid": false,
        "isValidating": false,
        "submitCount": 0,
        "touched": Object {},
        "validateOnBlur": true,
        "validateOnChange": true,
        "values": Object {
          "activities": Array [
            Object {
              "actionCode": "Case Confirmation",
              "actionDescription": "Confirm the index case",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Case Confirmation",
              "goalDescription": "Confirm the index case",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 1,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "RACD Register Family",
              "actionDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Family Registration",
              "goalDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 100,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "Blood Screening",
              "actionDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Blood screening",
              "goalDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 100,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "Bednet Distribution",
              "actionDescription": "Visit 100% of residential structures in the operational area and provide nets",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Bednet Distribution",
              "goalDescription": "Visit 100% of residential structures in the operational area and provide nets",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 100,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "Larval Dipping",
              "actionDescription": "Perform a minimum of three larval dipping activities in the operational area",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Larval Dipping",
              "goalDescription": "Perform a minimum of three larval dipping activities in the operational area",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 3,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "Mosquito Collection",
              "actionDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Mosquito Collection",
              "goalDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 3,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
            Object {
              "actionCode": "BCC",
              "actionDescription": "Conduct BCC activity",
              "actionIdentifier": "",
              "actionReason": "Investigation",
              "actionTitle": "Behaviour Change Communication",
              "goalDescription": "Complete at least 1 BCC activity for the operational area",
              "goalDue": "2019-08-12T10:17:17.765Z",
              "goalPriority": "medium-priority",
              "goalValue": 1,
              "timingPeriodEnd": "2019-08-12T10:17:17.765Z",
              "timingPeriodStart": "2019-08-05T10:17:17.765Z",
            },
          ],
          "caseNum": "",
          "date": "2019-08-05T10:17:17.765Z",
          "end": "2019-08-25T10:17:17.765Z",
          "identifier": "",
          "interventionType": "FI",
          "jurisdictions": Array [
            Object {
              "id": "",
              "name": "",
            },
          ],
          "name": "",
          "start": "2019-08-05T10:17:17.765Z",
          "status": "draft",
          "title": "",
          "version": "1",
        },
      }
    }
    id="start"
    inlineFocusSelectedMonth={false}
    monthsShown={1}
    nextMonthButtonLabel="Next month"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onInputClick={[Function]}
    onInputError={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    preventOpenOnFocus={false}
    previousMonthButtonLabel="Previous Month"
    readOnly={false}
    renderDayContents={[Function]}
    required={true}
    selected={2019-08-05T12:12:25.670Z}
    shouldCloseOnSelect={true}
    showMonthYearPicker={false}
    showTimeInput={false}
    showTimeSelect={false}
    strictParsing={false}
    timeCaption="Time"
    timeInputLabel="Time"
    timeIntervals={30}
    type="date"
    withPortal={false}
  >
    <PopperComponent
      hidePopper={true}
      popperComponent={null}
      popperModifiers={
        Object {
          "preventOverflow": Object {
            "boundariesElement": "viewport",
            "enabled": true,
            "escapeWithReference": true,
          },
        }
      }
      popperPlacement="bottom-start"
      popperProps={Object {}}
      targetComponent={
        <div
          className="react-datepicker__input-container"
        >
          <input
            className="form-control"
            disabled={false}
            id="start"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            required={true}
            type="text"
            value="2019-08-05"
          />
        </div>
      }
    >
      <Manager>
        <Reference>
          <InnerReference
            setReferenceNode={[Function]}
          >
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  className="form-control"
                  disabled={false}
                  id="start"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  readOnly={false}
                  required={true}
                  type="text"
                  value="2019-08-05"
                />
              </div>
            </div>
          </InnerReference>
        </Reference>
      </Manager>
    </PopperComponent>
  </DatePicker>
</DatePickerWrapper>
`;
